modules/api/pkg/resources/machine/common.go

```diff
@@ -273,30 +274,48 @@ func getVSphereProviderSpec(c *kubermaticv1.Cluster, nodeSpec apiv1.NodeSpec, dc
 	return EncodeAsRawExtension(config)
 }
 
-func getBaremetalProviderSpec(cluster *kubermaticv1.Cluster, nodeSpec apiv1.NodeSpec, dc *kubermaticv1.Datacenter) (*runtime.RawExtension, error) {
-	// Check if Tinkerbell configuration is provided
+func GetBaremetalProviderConfig(cluster *kubermaticv1.Cluster, nodeSpec apiv1.NodeSpec, dc *kubermaticv1.Datacenter) (*baremetal.RawConfig, error) {
+	// Ensure Tinkerbell configuration is provided
 	if nodeSpec.Cloud.Baremetal.Tinkerbell == nil {
-		return nil, errors.New("tinkerbell provisioner is required")
+		return nil, errors.New("tinkerbell provisioner is required in baremetal configuration")
 	}
```

Comment:
What if `nodeSpec.Cloud.Baremetal` is nil? This would result in a null exception. Please update this condition.

Reply:
we are checking this field before calling the function.